components/b2b/OrderSummaryEditable/OrderSummaryEditable.tsx

```diff
@@ -0,0 +1,435 @@
+  }
+
+  return (
+    <List
+      sx={{ width: '100%', maxWidth: 860 }}
+      component="nav"
+      aria-labelledby="order-summary-editable"
+    >
+      <ListItem
+        disableGutters
+        secondaryAction={
+          <Box display="flex" gap={2}>
+            {isEdit ? (
+              <>
+                <Button
+                  variant="contained"
+                  color="secondary"
+                  size="small"
+                  onClick={handleEditButtonClick}
+                >
+                  {t('cancel')}
+                </Button>
+                <Button variant="contained" color="primary" size="small" onClick={handleSave}>
+                  {t('save')}
+                </Button>
+              </>
+            ) : (
+              <Button
+                variant="contained"
+                color="secondary"
+                size="small"
+                onClick={handleEditButtonClick}
+              >
+                {t('edit')}
+              </Button>
+            )}
+          </Box>
+        }
+      >
+        <ListItemText
+          primary={
+            <Typography variant="subtitle1" fontWeight={'bold'}>
+              {t('summary')}
+            </Typography>
+          }
+        />
+      </ListItem>
+      <SummarySection
+        title="Item"
+        total={itemTotal}
+        subtotal={subTotal}
+        adjustment={adjustment}
+        taxTotal={itemTaxTotal}
+        isEdit={isEdit}
+        adjustmentValue={adjustmentValue.adjustment}
+        setAdjustmentValue={(val) => handleAdjustmentValue(val, 'adjustment')}
+      />
+      <SummarySection
+        title="Shipping"
+        total={shippingTotal}
+        subtotal={shippingSubTotal}
+        adjustment={shippingAdjustment}
+        taxTotal={shippingTaxTotal}
+        isEdit={isEdit}
+        adjustmentValue={adjustmentValue.shippingAdjustment}
+        setAdjustmentValue={(val) => handleAdjustmentValue(val, 'shippingAdjustment')}
+      />
+      <SummarySection
+        title="Handling"
+        total={handlingTotal}
+        subtotal={handlingSubTotal}
+        adjustment={handlingAdjustment}
+        taxTotal={handlingTaxTotal}
+        isEdit={isEdit}
+        adjustmentValue={adjustmentValue.handlingAdjustment}
+        setAdjustmentValue={(val) => handleAdjustmentValue(val, 'handlingAdjustment')}
+      />
+      <ListItem
+        slotProps={{
+          root: {
+            'aria-label': 'duty-total',
+          },
+        }}
+        sx={{ paddingLeft: 0 }}
+        secondaryAction={
+          <Price
+            fontWeight="normal"
+            variant="body2"
+            price={t('currency', { val: dutyTotal?.toString() })}
+          />
+        }
+      >
+        <ListItemIcon sx={{ minWidth: 30 }} />
+        <ListItemText primary={<Typography variant="body2">{t('duty-total')}</Typography>} />
+      </ListItem>
+    </List>
+  )
+}
+
+export default OrderSummaryEditable
```

Comment:
Separate the component in different file.

Reply:
Done